ores/lock_manager/poolcounter.py

```diff
@@ -0,0 +1,53 @@
+"""
+Class to implement PoolCounter lock manager.
+
+This is useful to make sure too many connections are not coming from one IP.
+"""
+
+import socket
+
+from .lock_manager import LockManager
+
+
+class PoolCounter(LockManager):
+    def __init__(self, config):
+        self.server = config['server']
+        self.port = config['port']
+        self.stream = None
+
+    def connect(self):
+        self.stream = socket.socket(socket.AF_INET, socket.SOCK_STREAM, 0)
+        return self.stream.connect((self.server, self.port))
+
+    def lock(self, key, workers, maxqueue, timeout):
+        if not self.stream:
+            self.connect()
+        try:
+            self.stream.send(bytes('ACQ4ME %s %d %d %d\n' % (key, workers, maxqueue, timeout), 'utf-8'))
+            data = self.stream.recv(4096).decode('utf-8')
+        except socket.error as e:
+            self.stream = None
```

Comment:
Maybe we need to disconnect to avoid using up sockets?

Reply:
I think the best solution would be to use socket.socket as a context manager rather than storing as a class member.

Reply:
Done